mos_tests/ironic/actions.py

```diff
@@ -0,0 +1,129 @@
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+from ironicclient import client
+
+from mos_tests.functions import common
+from mos_tests import settings
+
+
+class IronicActions(object):
+    """Ironic-specific actions"""
+
+    def __init__(self, os_conn):
+        self.os_conn = os_conn
+        token = os_conn.keystone.auth_token
+        ironic_endpoint = os_conn.keystone.service_catalog.url_for(
+            service_type='baremetal',
+            endpoint_type='publicURL')
+        self.client = client.get_client(api_version=1,
+                                        os_auth_token=token,
+                                        ironic_url=ironic_endpoint)
+
+    def _get_image(self, name):
+        return self.os_conn.nova.images.find(name=name)
+
+    def get_provisioned_node(self):
+        """Return ironic node wich have non zero vcpu in hypervisor
+
+        :rtype: ironicclient.v1.node.Node
+        :return: provisioned ironic node
+        """
+        for node in self.client.node.list():
+            try:
+                hypervisor = self.os_conn.nova.hypervisors.find(
+                    hypervisor_hostname=node.uuid)
+            except Exception:
+                continue
+            if hypervisor.vcpus > 0:
+                return node
+
+    def boot_instance(self, image, flavor, keypair, **kwargs):
+        """Boot and return ironic instance
+
+        :param os_conn: initialized `os_conn` fixture
+        :type os_conn: mos_tests.environment.os_actions.OpenStackActions
+        :param image: image to boot instance with it
+        :type image: warlock.core.image
+        :param flavor: baremetal flavor
+        :type flavor: novaclient.v2.flavors.Flavor
+        :param keypair: SSH keypair to instance
+        :type keypair: novaclient.v2.keypairs.Keypair
+        :return: created instance
+        :rtype: novaclient.v2.servers.Server
+        """
+        common.wait(self.get_provisioned_node,
+                    timeout_seconds=2 * 60,
+                    sleep_seconds=15,
+                    waiting_for='ironic node to be provisioned')
+        baremetal_net = self.os_conn.nova.networks.find(label='baremetal')
+        return self.os_conn.create_server('ironic-server',
+                                          image_id=image.id,
+                                          flavor=flavor.id,
+                                          key_name=keypair.name,
+                                          nics=[{'net-id': baremetal_net.id}],
+                                          timeout=60 * 10,
+                                          **kwargs)
+
+    def create_node(self, baremetal_node, server_ssh_credentials):
```

Comment:
@kromanenko good idea. As i known, you use yaml configuration file for IPMI in fuel-qa. Can you provide an example of configuration (only structure)?